### PR TITLE
Feature: Add CircleCI and CodeCov

### PR DESCRIPTION
**Pull request by angeldroth**
More robust testing framework
(Repo owners will need to sign up to free tier etc)

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -112,3 +112,4 @@ venv.bak/
 *.pkl
 
 .DS_Store
+.vscode
```
Comment (mzjp2) [inline .gitignore:115]:
```suggestion
.vscode

```

😅

Comment (angeldroth) [reply .gitignore:115]: hehe nice catch again

**Comment by angeldroth** [comment]
@mzjp2 obvs override anything I do here, your xp in this area is infinitely greater than mine haha

**Comment by mzjp2** [comment]
By the way, any specific reason for CCI v/s GH Actions? I see GH actions are kinda already used here and this repo integrates with GH pages - might make sense to stay within the GH ecosystem? (just a random thought)

**Comment by alejandroschuler** [comment]
@mzjp2 yes, that was my original thought when I set up GH actions. Unless there are very specific things that GH doesn't provide I'd like to stick with it since it seems simpler. Open to different opinions, though- as I've said, I'm not a software engineer and I'll do anything to avoid being one :P

**Comment by angeldroth** [comment]
Both fair points, I've just gotten used to CCI and it's expanded AET of features, but GHA might be better here for simplicity. Feel free to close PR if not needed :)

**Comment by mzjp2** [comment]
I've overhauled the workflow for this library in https://github.com/stanfordmlgroup/ngboost/pull/204 @angeldroth - but done nothing about code coverage. If that gets accepted, then maybe you can build on top of that to add codecov (or even pytest-cov tbh)

**Pull request closed without merging**